pkg/prom/prom.go

```diff
@@ -374,6 +378,21 @@ type PrometheusClientConfig struct {
 
 // NewPrometheusClient creates a new rate limited client which limits by outbound concurrent requests.
 func NewPrometheusClient(address string, config *PrometheusClientConfig) (prometheus.Client, error) {
+
+	var tlsCaCert *x509.CertPool
+	// get the bearer token and ca cert from serviceaccount if we have kube rbac proxy enabled
+	if env.IsKubeRbacProxyEnabled() {
+		restConfig, err := restclient.InClusterConfig()
+		if err != nil {
+            log.Errorf("ENABLE_KUBE_RBAC_PROXY was set to true but failed to get in-cluster config: %s", err)
+		}
+		config.Auth.BearerToken = restConfig.BearerToken
+		tlsCaCert, err = certutil.NewPool(`/var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt`)
+		if err != nil {
+            log.Errorf("ENABLE_KUBE_RBAC_PROXY was set to true but failed to load service-ca.crt: %s", err)
```

Comment:
Open question. If we are unable to get the cluster's CA cert but the user specifies `ENABLE_KUBE_RBAC_PROXY=true`, should we gracefully exit the program here? In other words, should we be returning the error here as opposed to just logging it and moving on?

Reply:
Exiting the program due to this will make it unusable if the user is not able to get the cluster CA cert for some reason, while there is a way that the user will still be able to successfully deploy kubecost or opencost if this happens. It can be possible that the user sets `INSECURE_SKIP_VERIFY` env var to be true, this will make the ca cert value not needed.